arangod/RocksDBEngine/RocksDBV8Functions.cpp

```diff
@@ -221,6 +223,86 @@ static void JS_CollectionRevisionTreeCorrupt(v8::FunctionCallbackInfo<v8::Value>
   TRI_V8_RETURN_UNDEFINED();
   TRI_V8_TRY_CATCH_END
 }
+
+static void JS_CollectionRevisionTreeVerification(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+
+  auto* collection = UnwrapCollection(isolate, args.Holder());
+
+  if (!collection) {
+    TRI_V8_THROW_EXCEPTION_INTERNAL("cannot extract collection");
+  }
+
+  std::unique_ptr<containers::RevisionTree> storedTree;
+  std::unique_ptr<containers::RevisionTree> computedTree;
+
+  {
+    TRI_vocbase_t& vocbase = collection->vocbase();
+    auto& server = vocbase.server();
+    RocksDBEngine& engine = server.getFeature<EngineSelectorFeature>().engine<RocksDBEngine>();
+    RocksDBReplicationManager* manager = engine.replicationManager();
+    double ttl = 3600;
+    RocksDBReplicationContext* ctx 
+      = manager->createContext(engine, ttl, SyncerId{17}, ServerId{17}, "");
+    RocksDBReplicationContextGuard guard(manager, ctx);
```

Comment:
I think `ctx` can be a nullptr here. We should check the return value and throw an exception if it is a nullptr.

Reply:
Done.